internal/evmreader/evmreader.go

```diff
@@ -25,7 +26,7 @@ import (
 type InputSource interface {
 	// Wrapper for FilterInputAdded(), which is automatically generated
 	// by go-ethereum and cannot be used for testing
-	RetrieveInputs(opts *bind.FilterOpts, appContract []common.Address, index []*big.Int,
+	RetrieveInputs(opts *bind.FilterOpts, appAddresses []common.Address, index []*big.Int,
```

Comment:
Please use the type alias from the `model` package. The same goes for the other files.

```suggestion
	RetrieveInputs(opts *bind.FilterOpts, appAddresses []Address, index []*big.Int,
```